test-plan/eventing/channel-lifecycle-conformance.md

```diff
@@ -0,0 +1,229 @@
+```
+
+Tested in eventing:
+- https://github.com/knative/eventing/blob/release-0.26/test/rekt/features/channel/control_plane.go#L102
+- https://github.com/knative/eventing/blob/release-0.26/test/conformance/channel_status_test.go#L29
+
+### [Output]
+
+```
+{
+  "test": "control-plane/channel-lifecycle/channel-addressable"
+  "output": {
+  	"channelImplementation": "",
+	  "obtainedURL": "<CHANNEL URL>"
+  }
+}
+```
+
+## [Pre] Create Subscription for the Channel
+
+First lets create a Service that works as a Subscriber and a deadLetterSink for the Subscription:
+
+```
+kubectl apply -f control-plane/channel-lifecycle/services.yaml
+```
+
+Create a Subscription that points to the Channel:
+
+```
+kubectl apply -f control-plane/channel-lifecycle/subscription.yaml
+```
+
+## [Test] Channel Reference in Subscription
+
+Check that the Subscription is making a reference to the Channel, this should return the name of the Channel.
+
+```
+kubectl get subscription conformance-subscription -ojsonpath="{.spec.channel.name}"
+```
+
+Tested in eventing:
+- https://github.com/knative/eventing/blob/release-0.26/test/rekt/resources/subscription/subscription_test.go#L166
+
+### [Output]
+
+```
+{
+  "test": "control-plane/channel-lifecycle/channel-reference-in-subscription"
+  "output": {
+  	"channelImplementation": "<CHANNEL IMPLEMENTATION>",
+	  "expectedReference": "conformance-channel"
+  }
+}
+```
+
+## [Test] Subscription for Channel Readiness
+
+Check for condition type `Ready` with status `True`: 
+
+```
+kubectl get subscription conformance-subscription -ojsonpath="{.status.conditions[?(@.type == \"Ready\")].status}"
+```
+
+Tested in eventing:
+- https://github.com/knative/eventing/blob/release-0.26/test/rekt/features/channel/control_plane.go#L84
+- https://github.com/knative/eventing/blob/release-0.26/test/conformance/channel_status_subscriber_test.go#L29
+
+### [Output]
+
+```
+{
+  "test": "control-plane/channel-lifecycle/subscription-for-channel-readiness"
+  "output": {
+  	"channelImplementation": "<CHANNEL_IMPLEMENTATION>",
+    "expectedType": "Ready",
+    "expectedStatus": "True"
+  }
+}
+```
+
+### [Test] Channel fan out message to Subscribers
+
+Test for messages sent from the channel to each Subscription Subscriber:
+
+Lets create first some Ping Sources to start sending events to the conformance-channel:
+
+
+```
+kubectl apply -f control-plane/channel-lifecycle/ping-sources.yaml
+```
+
+Now lets look for those events in each Subscription Subscriber ref logs:
+
+```
+kubectl logs --ignore-errors --tail 100 -l serving.knative.dev/service=conformance-events -c user-container | grep conformance-pingsource-1 | tail -n 5
+
+kubectl logs --ignore-errors --tail 100 -l serving.knative.dev/service=conformance-events -c user-container | grep conformance-pingsource-2 | tail -n 5
+
+kubectl logs --ignore-errors --tail 100 -l serving.knative.dev/service=conformance-events -c user-container | grep conformance-pingsource-3 | tail -n 5
+```
```

Comment:
Lionel suggested that I use the `recordevents` image for sending and receiving events, which I did here (in-review). You can specify how many events you want to send and how many you want to see received, which seems to fit nicely this test.

Receive:
https://github.com/knative/specs/blob/1048de5cb5a4e1a74cf8c763bf53efbf6f003572/test-plan/eventing/control-plane/event-delivery/01-00-channel-ack.yaml

Send:
https://github.com/knative/specs/blob/1048de5cb5a4e1a74cf8c763bf53efbf6f003572/test-plan/eventing/control-plane/event-delivery/01-01-channel-ack.yaml

It got me some time to learn the ways of `recordevents`, ping me if you consider using it and need help.